dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_BasicsTests/AutoScaleMethodsTests.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier:  Apache-2.0
+
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using AutoScale_Basics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Amazon.AutoScaling;
+using Amazon.AutoScaling.Model;
+
+namespace AutoScale_Basics.Tests
+{
+    [TestClass()]
+    public class AutoScaleMethodsTests
+    {
+        private readonly string _GroupName = "test-group-name";
+        private readonly AmazonAutoScalingClient _Client = new AmazonAutoScalingClient();
+
+        // the Amazon Resource Name (ARN) of the IAM service linked role.
+        private readonly string _ServiceLinkedRoleArn = "<Enter Value>";
+
+        private readonly string _LaunchTemplateName = "AutoScaleLaunchTemplateTest";
+
+        [TestMethod()]
+        public async Task DeleteAutoScalingGroupTest()
+        {
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, _GroupName);
+            Assert.IsTrue(success, "Could not delete the group.");
+        }
+
+        [TestMethod()]
+        public async Task DeleteAutoScalingGroupGroupNonexistentTest()
+        {
+            var success = await AutoScaleMethods.DeleteAutoScalingGroupAsync(_Client, "NonExistentSite");
+            Assert.IsFalse(success, "Could not find the group to delete.");
```

Comment:
I still think this failure message is wrong. The action here is that you give it a bad group, "NonExistentSite," and try to delete it. You assert false, which is correct, but that means it **did not** delete the group: it successfully errored on trying to delete "NonExistentSite."
If you get a true, then the assert failed, in which case you **successfully** deleted a group that should not exist. So your failure message for the assert should be something like:
"Deleted a non existent group!"

Reply:
Make the change. Thanks for insisting because you were right. It made sense.